atc/db/dbtest/builder.go

```diff
@@ -77,6 +77,25 @@ func (builder Builder) WithWorker(worker atc.Worker) SetupFunc {
 	}
 }
 
+func (builder Builder) WithTeamWorker(worker atc.Worker) SetupFunc {
+	return func(scenario *Scenario) error {
+		if scenario.Team == nil {
+			err := builder.WithTeam(unique("team"))(scenario)
```

Comment:
Should this use `worker.Team`? Or, should we just error if the team is unset?

Also, might it make sense for `WithWorker` to just handle the case where `worker.Team` is set, and find the appropriate team to save the worker to? I'm not sure if there's a case where `worker.Team` is set but you don't want to `SaveWorker`

Reply:
Yeah good call. I pushed https://github.com/concourse/concourse/pull/6156/commits/bded4c77fa3fdfa852f3f3eb185cb91b2f803dfc which removes `WithTeamWorker` and adjusts `WithWorker` to respect the `Team` field instead.